consensus/polybft/consensus_runtime.go

```diff
@@ -214,34 +221,52 @@ func (c *consensusRuntime) AddLog(eventLog *ethgo.Log) {
 
 // OnBlockInserted is called whenever fsm or syncer inserts new block
 func (c *consensusRuntime) OnBlockInserted(block *types.Block) {
+	c.lock.Lock()
```

Comment:
Could we just use the `getLastBuiltBlockAndEpoch` function to read the last built block and `epochMetadata`, and then just lock the lines where we set the, `c.epoch`, `c.lastBuiltBlock` and `c.proposerSnapshot`?

Reply:
No, we need everything to be executed in exactly one transaction. Everything must go under the lock :(